File: solutions/python/grains/1/grains.py
def square(number):
    if number>64 or number<1:
        raise ValueError("square must be between 1 and 64")
    return 2**(number-1)
    
def total():
    grains=0
    for i in range(64):
        grains+=2**(i)
    return int(grains)

File: solutions/typescript/grains/1/grains.ts
export const square = (square: number): bigint => {
  if (square < 1 || square > 64){
    throw new Error('square must be between 1 and 64')
  }
  return 2n ** (BigInt(square) - 1n);
};

export const total = (): bigint => {
  return 2n**64n - 1n;
};
